enhancements/operator-dev-doc.md

```diff
@@ -0,0 +1,269 @@
+
+The [Custom Resource Definitions](https://kubernetes.io/docs/concepts/extend-kubernetes/api-extension/custom-resources/#customresourcedefinitions) in OpenShift can be viewed with this:
+```console
+$ oc get crds
+
+For details on a particular CRD (ex: openshiftapiservers.operator.openshift.io):
+$ oc explain openshiftapiservers.operator.openshift.io
+```
+For more information, check out [the Operator pattern in Kubernetes](https://kubernetes.io/docs/concepts/extend-kubernetes/operator/).    
+
+While ClusterOperators manage individual components (Custom Resources), the 
+[Cluster Version Operator](https://github.com/openshift/cluster-version-operator) (CVO) is responsible for managing the ClusterOperators.
+
+### What is an Operand?
+The component that an operator manages is its operand.  For example, the cluster-kube-controller-manager-operator CO manages the 
+cluster-kube-controller-manager component running in OpenShift. An operator can be responsible for multiple components.  For example, 
+the CVO manages all of the COs (in this way ClusterOperators are also operands). 
+
+### What is an OpenShift release image?
+An OpenShift release image is comprised of layers.  Each layer digest is a single component image.  This includes operator
+images, operand images, and others. To get a list of the components and their images that comprise a release payload, you can grab a 
+release from the [openshift release page](https://openshift-release.svc.ci.openshift.org/) and run:
+```console
+$ oc adm release info registry.svc.ci.openshift.org/ocp/release:version
+```
+
+If the above command fails, you need to authenticate against `registry.svc.ci.openshift.org`. 
+Add the necessary credentials to your local `~/.docker/config.json` (or equivalent file) like so:
+ - visit `https://api.ci.openshift.org`, `upper right corner '?'` dropdown to `Command Line Tools`
+ - copy the given `oc login https://api.ci.openshift.org --token=<hidden>`, paste in your terminal
+ - then run `oc registry login` to add your credentials to your containers tool config file     
+
+You'll notice that currently the release payload is just shy of 100 images.
+
+### CVO Cluster Operator Status
+You can check on the status of any ClusterOperator with
+```console
+$ oc get co/clusteroperatorname -o yaml
+```
+At any time the staus of a CO may be:
+- Available
+- Degraded
+- Progressing       
+
+[CVO Cluster Operator Status Conditions Overview](https://github.com/openshift/cluster-version-operator/blob/master/docs/dev/clusteroperator.md#conditions)
+
+### Metrics
+By default, an OpenShift ClusterOperator exposes [Prometheus](https://prometheus.io) metrics via `metrics` service.
+
+### Debugging
+Operators expose events that can help in debugging issues. To get operator events, run following command:
+
+```
+$ oc get events -n [cluster-operator-namespace]
+```
+
+### How do I build|update|verify|run unit-tests
```

Comment:
Is this fair to include in openshift/enhacements when not all operators use that library??

Reply:
thinking is to keep this doc here for now.  we will link to it from the operators/components for which it is relevant, won't promote it otherwise.